odl/contrib/fom/test/test_supervised.py

```diff
@@ -14,16 +14,46 @@
 import scipy.signal
 import scipy.misc
 import odl
-import odl.contrib.fom
-from odl.contrib.fom.util import filter_image_sep2d
-from odl.util.testutils import simple_fixture
+from odl.contrib import fom
+from odl.util.testutils import simple_fixture, noise_element
 
 fft_impl = simple_fixture('fft_impl',
                           [odl.util.testutils.never_skip('numpy'),
                            odl.util.testutils.skip_if_no_pyfftw('pyfftw')])
+
 space = simple_fixture('space',
                        [odl.rn(3),
-                        odl.uniform_discr(0, 1, 10)])
+                        odl.rn(3, dtype='float32'),
+                        odl.uniform_discr(0, 1, 10),
+                        odl.uniform_discr([0, 0], [1, 1], [5, 5])])
+
+scalar_fom = simple_fixture('FOM',
+                            [fom.mean_squared_error,
+                             fom.mean_absolute_error,
+                             fom.mean_value_difference,
+                             fom.standard_deviation_difference,
+                             fom.range_difference,
+                             fom.blurring])
+
+
+def test_general(space, scalar_fom):
+    """Test general properties of FOMs."""
+    for name, ground_truth in space.examples:
+        ground_truth = np.abs(ground_truth)
+        scale = np.max(ground_truth) - np.min(ground_truth)
+        noise = scale * np.abs(noise_element(space))
+        data = ground_truth + noise
+
+        # Check that range is a real number
+        assert np.isscalar(scalar_fom(data, ground_truth))
+
+        # Check that FOM is minimal when ground truth is compared with itself
+        assert (scalar_fom(ground_truth, ground_truth) <=
+                scalar_fom(data, ground_truth))
+
+        # Check that FOM is monotonic wrt noise level
+        assert (scalar_fom(ground_truth + noise, ground_truth) <=
+                scalar_fom(ground_truth + 2 * noise, ground_truth))
```

Comment:
This test fails for both `rn(3)` spaces, for both `standard_deviation_difference` and `range_difference`. If increased to `4  *  noise` is seems to stop failing.

The input `scalar_fom` is a function. Is there an easy way to check which function it is? Then we could exclude these two. @kohr-h or @adler-j?

Reply:
You can add a `print` statement in the test and run it again. `pytest` will capture the output on `stdout` and show it *if the test fails*. So don't forget to remove the statement again, it's easy to forget since `pytest` won't show the message any more when the test stops failing.

Reply:
But on the larger issue: maybe it's better to be a bit more generous instead of not testing certain FOMs.